jwst/ami/oifits.py

```diff
@@ -0,0 +1,763 @@
+#! /usr/bin/env python
+
+"""
+
+RawOifits class: takes fringefitter class, which contains nrm_list and instrument_data attributes,
+all info needed to write oifits.
+populate structure needed to write out oifits files according to schema.
+averaged and multi-integration versions, sigma-clipped stats over ints
+
+CalibOifits class: takes two AmiOIModel datamodels and produces a final calibrated datamodel.
+"""
+import numpy as np
+
+from scipy.special import comb
+from astropy import units as u
+from astropy.stats import sigma_clipped_stats
+from astropy.time.core import Time
+from astroquery.simbad import Simbad
+from astropy.coordinates import SkyCoord
+
+from stdatamodels.jwst import datamodels
+
+
+class RawOifits:
+    def __init__(self, fringefitter, nh=7, angunit="radians", method='median'):
+        """
+        Short Summary
+        -------------
+        Class to store AMI data in the format required to write out to OIFITS files
+        Based on ObservablesFromText from ImPlaneIA.
+        Angular quantities of input are in radians from fringe fitting; converted to degrees for saving.
+
+        Parameters
+        ----------
+        fringefitter: FringeFitter object
+            Object containing nrm_list attribute (list of nrm objects) 
+            and other info needed for OIFITS files
+        nh: integer
+            default 7, number of holes in the NRM
+        angunit: string
+            Unit of incoming angular quantities (radians or degrees). 
+            Will be converted to degrees for OIFITS. Default radians
+        method: string
+            Method to average observables: mean or median. Default median.
+
+
+        """
+        self.ff = fringefitter
+        self.nh = nh # 7
+        self.nslices = len(self.ff.nrm_list) # n ints
+        self.nbl = int(comb(self.nh, 2)) # 21
+        self.ncp = int(comb(self.nh, 3)) # 35
```

Comment:
I'd like to improve the readability of these variables. I'm assuming `nbl` is `n_baselines`. Is this correct? Also, what is `ncp` and `nca`? This comment applies to many variables throughout this PR.

Reply:
I think there is a bit of a history to the short variable names here; one of the original authors learned to code in C on punch cards where character counts were stricter, and is now not comfortable with a lot of typing for physical reasons. Nevertheless I can see the advantage of more descriptive variable names; I'm a bit afraid of breaking things by doing large-scale renaming at this point though...

Reply:
Thanks for the context.

I think the renaming will be easier to do as part of this PR (as the code is "fresh" in mind and testing is ongoing) rather than at some later date. It will help for later maintenance, debugging, and modifications. I must admit that I am not an expert when it comes to the algorithms used here but I also expect the same is true for some of the folks that will maintain this code.

I am happy to help with these changes with some guidance on what names to replace. Would it be helpful if I added more comments to variable names that are candidates for renaming?

Reply:
If your coding environment has a more reliable way of updating all usages of a given variable name than doing find/replace in each file that would be helpful; I'm not sure how to do that efficiently in the editor I'm using. 
In the lines you highlighted above, nbl is indeed n_baselines. nh = n_holes, ncp = n_closurephases, nca = n_closureamplitudes, fp = fringephases, fa = fringeamplitudes. Those are the ones that I saw immediately as good candidates for renaming, but if you find others that it would be helpful to have my input on let me know!
As always, thanks for your help in making this code a bit more readable!